File: api/models/ingestion.py
from __future__ import annotations

from typing import Any

from pydantic import BaseModel


class IngestionSource(BaseModel):
    name: str
    kind: str
    fresh: str
    target: str
    cadence: str
    rows: int
    last_error: str
    on: bool


class FeatureFreshness(BaseModel):
    name: str
    fresh: str
    kind: str
    rows: int


class IngestionResponse(BaseModel):
    sources: list[IngestionSource]
    features: list[FeatureFreshness]
    snapshot_lag: str
    snapshot_count: int
    schemas_ok: bool
    leakage_tests_ok: bool


class IngestionJobRequest(BaseModel):
    action: str
    limit: int = 100
    max_pages: int = 5
    include_closed: bool = True
    closed_only: bool = False
    include_trades: bool = True
    timeout_sec: int = 300


class IngestionJobStep(BaseModel):
    command: list[str]
    exit_code: int
    stdout: str
    stderr: str
    summary: Any | None = None


class IngestionJobResponse(BaseModel):
    action: str
    ok: bool
    steps: list[IngestionJobStep]

File: console/src/types/ingestion.ts
export interface IngestionSource {
  name: string
  kind: string
  fresh: string
  target: string
  cadence: string
  rows: number
  last_error: string
  on: boolean
}

export interface FeatureFreshness {
  name: string
  fresh: string
  kind: string
  rows: number
}

export interface IngestionResponse {
  sources: IngestionSource[]
  features: FeatureFreshness[]
  snapshot_lag: string
  snapshot_count: number
  schemas_ok: boolean
  leakage_tests_ok: boolean
}

export type IngestionAction =
  | 'migrate'
  | 'polymarket-cs'
  | 'polymarket-closed'
  | 'polymarket-price-history'
  | 'polymarket-account-history'
  | 'polymarket-reconcile'
  | 'polymarket-full-refresh'

export interface IngestionJobRequest {
  action: IngestionAction
  limit?: number
  max_pages?: number
  include_closed?: boolean
  closed_only?: boolean
  include_trades?: boolean
  timeout_sec?: number
}

export interface IngestionJobStep {
  command: string[]
  exit_code: number
  stdout: string
  stderr: string
  summary: unknown
}

export interface IngestionJobResponse {
  action: IngestionAction
  ok: boolean
  steps: IngestionJobStep[]
}
